R/FilterState.R

```diff
@@ -250,129 +187,79 @@ InteractiveFilterState <- R6::R6Class( # nolint
     },
 
     #' @description
-    #' Set whether to keep NAs.
+    #' Sets filtering state.
     #'
-    #' @param value `logical(1)`\cr
-    #'   value(s) which come from the filter selection. Value is set in `server`
-    #'   modules after selecting check-box-input in the shiny interface. Values are set to
-    #'   `private$keep_na` which is reactive.
+    #' @param state a `teal_slice` object
     #'
-    #' @return NULL invisibly
+    #' @return `self` invisibly
     #'
-    set_keep_na = function(value) {
-      checkmate::assert_flag(value)
-      if (shiny::isolate(private$is_disabled())) {
-        warning("This filter state is disabled. Can not change keep NA.")
+    set_state = function(state) {
+      checkmate::assert_class(state, "teal_slice")
+      if (private$fixed) {
+        logger::log_trace("{ this filter state is fixed: { private$dataname } { private$varname }")
       } else {
-        private$keep_na(value)
-        logger::log_trace(
-          sprintf(
-            "%s$set_keep_na set for variable %s to %s.",
-            class(self)[1],
-            private$varname,
-            value
+        # Allow for enabling a filter state before altering state.
+        if (isTRUE(state$disabled)) private$disable()
+        if (isFALSE(state$disabled)) private$enable()
+
+        if (private$is_disabled()) {
+          logger::log_trace("{ this filter state is disabled: { private$dataname } { private$varname }")
```

Comment:
I would expect a warning here. At the very least a `log_warn`. Attempting to set state on a disabled filter suggests something is wrong and as an app/module developer, I'd like to know about that.

Reply:
will do :+1: